firefox_ui_harness/testcases/update.py

```
@@ -0,0 +1,133 @@
+
+        # The upgraded version should be identical with the version given by
+        # the update and we shouldn't have run a downgrade
+        check = self.marionette.execute_script("""
+          Components.utils.import("resource://gre/modules/Services.jsm");
+
+          return  Services.vc.compare(arguments[0], arguments[1]);
+        """, script_args=[info['build_post']['version'], info['build_pre']['version']])
+
+        self.assertGreaterEqual(check, 0, 'The version of the upgraded build is higher or equal')
+
+        # If a target version has been specified, check if it matches the updated build
+        if self.target_version:
+            self.assertEqual(info['build_post']['version'], self.target_version)
+
+        # The post buildid should be identical with the buildid contained in the patch
+        self.assertEqual(info['build_post']['buildid'], info['patch']['buildid'])
+
+        # If a target buildid has been specified, check if it matches the updated build
+        if self.target_buildid:
+            self.assertEqual(info['build_post']['buildid'], self.target_buildid)
+
+        # An upgrade should not change the builds locale
+        self.assertEqual(info['build_post']['locale'], info['build_pre']['locale'])
+
+        # Check that no application-wide add-ons have been disabled
+        self.assertEqual(info['build_post']['disabled_addons'],
+                         info['build_pre']['disabled_addons'])
+
+    def restore_config_files(self):
+        # Reset channel-prefs.js file if modified
+        try:
+            if hasattr(self, 'default_update_channel'):
+                path = self.default_update_channel['path']
+                self.logger.info('Restoring channel defaults for: {}'.format(path))
+                with open(path, 'w') as f:
+                    f.write(self.default_update_channel['content'])
+        except IOError:
+            self.logger.exception('Failed to reset the default update channel.')
```

If you aren't going to let the exception propagate, call "self.fail()" to ensure the test fails if that's what's intended here.